### PR TITLE
merge socket api

### PR DESCRIPTION
**Pull request by heapwolf**
this preserves history and merges in socket-api making it easier to sync the javascript api with the runtime.

### CODE REVIEW

**Comment by heapwolf** [comment]
CI is probably struggling with the fact that code is in `/api` now

**Review by chicoxyzzy — APPROVED** [review]

**Comment by chicoxyzzy** [comment]
docs generation fixes PR #121

File: api/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (jwerle) [inline api/.eslintrc:1]: We should move this to the root of the repo

**Review by jwerle — APPROVED** [review]

**Comment by jwerle** [comment]
I can follow up with a subsequent PR that can include the `api/` directory in the installation and a copy into the platform specific output path (`socket/` sitting sibling to `index.html|).

Folks should be able to do:

```js
import fs from 'socket/fs.js'
```

**Pull request merged**